R/crunchbox.R

```diff
@@ -41,6 +46,28 @@ crunchBox <- function (dataset, filters=crunch::filters(dataset), ...) {
     ## Add "where" after so that it no-ops if variablesFilter returns NULL (i.e. no filter)
     payload$where <- variablesFilter(dataset)
 
+    ## Add colors if they exist to the payload
+    if (!missing(brand_colors)) {
+        if (!is.list(brand_colors) ||
+            any(!names(brand_colors) %in% c("primary", "secondary", "message"))) {
+            halt(sQuote("brand_colors"), " must be a named list with only ",
+                 serialPaste(dQuote(c("primary", "secondary", "message")), collapse = "or"))
+        }
+        payload$display_settings$palette$brand_colors <- brand_colors
+    }
+    if (!missing(static_colors)) {
+        if (!is.list(static_colors)) {
+            halt(sQuote("static_colors"), " must be a list of characters")
+        }
+        payload$display_settings$palette$static_colors <- static_colors
+    }
+    if (!missing(category_color_lookup)) {
+        if (!is.list(category_color_lookup) || is.null(names(category_color_lookup))) {
```

Comment:
Do we want to try to check that the categories supplied in the list match categories in the data? Or if there are mismatches on either side?

Reply:
I think that might be too expensive (but perhaps something to add to the `preCrunchBoxCheck`?)

Reply:
I've added this as a todo in precrunchbox check for now.